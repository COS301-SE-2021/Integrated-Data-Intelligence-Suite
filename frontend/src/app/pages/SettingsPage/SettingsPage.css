
main .outer-container#settings-container{
    /*border: 2px solid ;*/

}

.ant-col>div.option{
    /*background-color: white;*/
    /*border-radius: 10px;*/
    margin: 0;
    padding: 10px;
    line-height: normal;
}

.ant-col>div.option.active{
    background: #5873f9;
    border-radius: 5px;
    color: white;
}

.bodyDiv .header{
    background: white;
    display: inline-flex;
    /*width: 100vw;*/
    align-items: end;
}

.top-padding{
    width: 100%;
    /*border: 3px solid red;*/
    height: 50px;
    /*padding-right: 100px;*/
}

.ant-col .left-column{
    padding-top: 50px;
}

.ant-col .right-column{
    background: #eff0f0;

}

.bodyDiv .ant-row{
    height: 100vh;
    flex-flow: unset;
}


.back-button{
    margin: 15px 15px 100px 0;
    font-size: 30px;
    float: right;
    position: relative;
}


.ant-col>div.ant-divider{
    border-color: #5873f9;
}
