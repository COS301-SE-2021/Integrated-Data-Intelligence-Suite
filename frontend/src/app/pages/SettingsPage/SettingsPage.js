import React, {useState} from 'react';
import {Layout, Row, Col, Divider} from 'antd';
import {CloseCircleTwoTone} from '@ant-design/icons'
import { useHistory} from "react-router-dom";
import Permissions from "../PermissionsPage/Permissions";
import ProfilePage from "../ProfilePage/ProfilePage";


const { Content} = Layout;

const setActive = (component) =>{
    let options = document.getElementsByClassName("option");
    console.log(options)
    for (let i = 0; i < options.length; i++) {
        console.log(options[i].id)
        if(options[i].id === component){
            options[i].className = "option active"
        }else{
            options[i].className = "option"
        }
    }
    return true;
}

function getLocalUser() {
    const localUser = localStorage.getItem("user");
    if (localUser) {
        // console.log("user logged in is ", localUser)
        return JSON.parse(localUser);
    } else {
        return null;
    }
}




const SettingsPage = () => {

    const [component, setComponent] = useState("profile");
    const [user, setUser] = useState(getLocalUser());
    const history = useHistory();


    return (
        <Layout className={"bodyDiv"}>
            <div className={"header white-background"}>
            </div>
                <Content id={"settings-container"} className={"outer-container"} style={{ margin: '0' , minHeight:"100vh"}}>
                    <Row className={"row"}>
                        <Col style={{padding:"30px 20px"}} className={"left-column"} flex="200px">
                            <div id={"Profile"} className={"active option"} onClick={()=>setComponent("Profile")}>Profile</div>
                            <Divider />

                            { user && user.isAdmin && <div id={"Permissions"} className={"option"} onClick={()=>setComponent("Permissions")}>Permissions</div>}
                            {user && user.isAdmin && <Divider />}

                            <div id={"Account"} className={"option"} onClick={()=>setComponent("Account")}>Account</div>
                        </Col>
                        <Col style={{padding:"0 0px 30px 0", backgroundColor:"#eff0f0"}} className={"right-column"} flex="auto">
                            {<div className={"top-padding"}><CloseCircleTwoTone className={"back-button"} onClick={()=>history.go(-1)}/></div>}
                            { component === "Permissions" && user && user.isAdmin && setActive(component) && <Permissions/>}
                            { component === "Profile" && setActive(component) && <div> <h1>Page not implemented</h1></div>}
                            { component === "Account" && setActive(component) && <div><h1>Page not implemented</h1></div>}
                        </Col>
                    </Row>
                </Content>
        </Layout>

    );
};
export default SettingsPage;
