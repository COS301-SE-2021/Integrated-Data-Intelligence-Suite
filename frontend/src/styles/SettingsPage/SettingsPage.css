.bodyDiv{
    height: 99vh;
}

main .outer-container#settings-container{
    border: 2px solid ;

}