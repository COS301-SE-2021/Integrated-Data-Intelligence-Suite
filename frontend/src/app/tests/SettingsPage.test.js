import React from 'react';
import renderer from 'react-test-renderer';
import SettingsPage from "../pages/SettingsPage/SettingsPage";

it('renders correctly', () => {
    const tree = renderer
        .create(<SettingsPage/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});