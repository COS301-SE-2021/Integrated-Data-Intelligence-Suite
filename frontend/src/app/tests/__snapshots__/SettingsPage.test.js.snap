// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="ant-layout bodyDiv"
>
  <div
    className="header white-background"
  />
  <main
    className="ant-layout-content outer-container"
    id="settings-container"
    style={
      Object {
        "margin": "0",
        "minHeight": "100vh",
      }
    }
  >
    <div
      className="ant-row row"
      style={Object {}}
    >
      <div
        className="ant-col left-column"
        style={
          Object {
            "flex": "0 0 200px",
            "padding": "30px 20px",
          }
        }
      >
        <div
          className="option"
          id="Profile"
          onClick={[Function]}
        >
          Profile
        </div>
        <div
          className="ant-divider ant-divider-horizontal"
          role="separator"
        />
        <div
          className="option"
          id="Account"
          onClick={[Function]}
        >
          Account
        </div>
      </div>
      <div
        className="ant-col right-column"
        style={
          Object {
            "backgroundColor": "#eff0f0",
            "flex": "auto",
            "padding": "0 0px 30px 0",
          }
        }
      >
        <div
          className="top-padding"
        >
          <span
            aria-label="close-circle"
            className="anticon anticon-close-circle back-button"
            onClick={[Function]}
            role="img"
            tabIndex={-1}
          >
            <svg
              aria-hidden="true"
              data-icon="close-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                fill="#1890ff"
              />
              <path
                d="M512 140c-205.4 0-372 166.6-372 372s166.6 372 372 372 372-166.6 372-372-166.6-372-372-372zm171.8 527.1c1.2 1.5 1.9 3.3 1.9 5.2 0 4.5-3.6 8-8 8l-66-.3-99.3-118.4-99.3 118.5-66.1.3c-4.4 0-8-3.6-8-8 0-1.9.7-3.7 1.9-5.2L471 512.3l-130.1-155a8.32 8.32 0 01-1.9-5.2c0-4.5 3.6-8 8-8l66.1.3 99.3 118.4 99.4-118.5 66-.3c4.4 0 8 3.6 8 8 0 1.9-.6 3.8-1.8 5.2l-130.1 155 129.9 154.9z"
                fill="#e6f7ff"
              />
              <path
                d="M685.8 352c0-4.4-3.6-8-8-8l-66 .3-99.4 118.5-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155-130.1 154.9a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3 99.3-118.5L611.7 680l66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.9 512.2l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
                fill="#1890ff"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </main>
</section>
`;
